src/Configuration/Dotenv.hs

```diff
@@ -12,95 +12,108 @@
 {-# LANGUAGE RecordWildCards #-}
 
 module Configuration.Dotenv
-  (
-  -- * Dotenv Load Functions
-    load
-  , loadFile
-  , parseFile
-  , onMissingFile
-  -- * Dotenv Types
-  , module Configuration.Dotenv.Types
-  )
- where
+    ( -- * Dotenv Load Functions
+      load
+    , loadFile
+    , parseFile
+    , onMissingFile
+      -- * Dotenv Types
+    , module Configuration.Dotenv.Types) where
 
 import           Configuration.Dotenv.Environment    (getEnvironment, lookupEnv,
                                                       setEnv)
 import           Configuration.Dotenv.Parse          (configParser)
 import           Configuration.Dotenv.ParsedVariable (interpolateParsedVariables)
-import           Configuration.Dotenv.Types          (Config (..),
-                                                      defaultConfig)
-import           Control.Monad                       (liftM)
+import           Configuration.Dotenv.Types          (Config (..), ReaderT, ask,
+                                                      defaultConfig,
+                                                      liftReaderT, runReaderT)
+import           Control.Monad                       (when)
 import           Control.Monad.Catch
 import           Control.Monad.IO.Class              (MonadIO (..))
 import           Data.List                           (intersectBy, union,
                                                       unionBy)
 import           System.IO.Error                     (isDoesNotExistError)
 import           Text.Megaparsec                     (errorBundlePretty, parse)
 
+-- | Monad Stack for the application
+type DotEnv m a = ReaderT Config m a
+
 -- | Loads the given list of options into the environment. Optionally
 -- override existing variables with values from Dotenv files.
 load ::
   MonadIO m =>
   Bool -- ^ Override existing settings?
   -> [(String, String)] -- ^ List of values to be set in environment
   -> m ()
-load override = mapM_ (applySetting override)
+load override kv = runReaderT (mapM_ applySetting kv) defaultConfig { configOverride = override }
 
 -- | @loadFile@ parses the environment variables defined in the dotenv example
 -- file and checks if they are defined in the dotenv file or in the environment.
 -- It also allows to override the environment variables defined in the environment
 -- with the values defined in the dotenv file.
 loadFile
-  :: MonadIO m
-  => Config -- ^ Dotenv configuration
-  -> m [(String, String)] -- ^ Environment variables loaded
-loadFile Config{..} = do
-  environment <- liftIO getEnvironment
-  readedVars <- concat `liftM` mapM parseFile configPath
-  neededVars <- concat `liftM` mapM parseFile configExamplePath
-  let coincidences = (environment `union` readedVars) `intersectEnvs` neededVars
+  :: MonadIO m => DotEnv m [(String, String)]
```

Comment:
@FranzGB Let's not change the signature of any of the exported functions. We can keep the previous signature by doing this:
```hs
loadFile config = runReader config $ do
  Config { .. } <- ask
  ... 
```

As we discussed internally, I want to hide the details of the `Dotenv` monad from the outside world because it's just an implementation detail: 

[Encapsulation](https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)) :wink:

Reply:
Also, I don't want to change the user interface just for an implementation detail.